test/unit/BuilderTest.php

```diff
@@ -564,15 +564,61 @@ public function withClaimMustKeepAFluentInterface()
     /**
      * @test
      *
+     * @param string $name
+     * @param mixed $value
+     * @param mixed $expected
+     *
      * @covers ::__construct
      * @covers ::withClaim
      * @covers \Lcobucci\JWT\Token\RegisteredClaimGiven
+     *
+     * @dataProvider dataWithClaimDeprecationNotice
      */
-    public function withClaimShouldThrowExceptionWhenTryingToConfigureARegisteredClaim()
+    public function withClaimShouldSendDeprecationNoticeWhenTryingToConfigureARegisteredClaim($name, $value, $expected)
+    {
+        $key = $this->createMock(Key::class);
+        $signature = $this->createMock(Signature::class);
+        $signature
+            ->expects(static::once())
+            ->method('hash')
+            ->willReturn('--hash--')
+        ;
+
+        $signer = $this->createMock(Signer::class);
+        $signer
+            ->expects(static::once())
+            ->method('sign')
+            ->willReturn($signature)
+        ;
+
+
+
+        $this->expectDeprecation('The use of the method "withClaim" is deprecated for registered claims. Please use dedicated method instead.');
+
+        $token = $this
+            ->createBuilder()
+            ->withClaim($name, $value)
+            ->getToken($signer, $key)
+        ;
+
+        self::assertEquals($expected, $token->claims()->get($name));
+    }
+
+    public function dataWithClaimDeprecationNotice()
     {
-        $this->expectException(RegisteredClaimGiven::class);
+        $now = time();
+        $nowAsDate = new DateTimeImmutable('@' . $now);
+        $nowPlus1HourAsDate = $nowAsDate->modify('+1 hour');
 
-        $this->createBuilder()->withClaim('sub', 'me');
+        return [
+            ['sub', 'me', 'me'],
+            ['aud', 'him', ['him']],
+            ['jti', '0123456789ABCDEF', '0123456789ABCDEF'],
+            ['iss', 'you', 'you'],
+            ['exp', $nowPlus1HourAsDate->getTimestamp(), $nowPlus1HourAsDate->getTimestamp()],
+            ['iat', $now, $nowAsDate->getTimestamp()],
+            ['nbf', $now, $nowAsDate->getTimestamp()],
```

Comment:
This is incorrect as we rely on objects for the new API.

Reply:
The problem is that the test fails when setting $nowAsDate

Reply:
It will fail until we add the other`if` statements in https://github.com/lcobucci/jwt/pull/561/files#diff-aaf0bc0443d3390951e65b1d6dadf6134554550b510609434969424757f1b6f9R418 (we need to forward the call to all specific methods for registered claims)

Reply:
Should be fine now